llvmcpy/_generator.py

```diff
@@ -874,35 +878,24 @@ def out_ptr(self):
             write(f"\n{enum_name} = {enum_values}\n")
 
 
-def initialize() -> str:
-    # Add to PATH the output of llvm-config --bin-dir
-    global search_paths
-    search_paths = os.environ.get("PATH", os.defpath)
-    global llvm_config
-    llvm_config = find_program("LLVM_CONFIG", ["llvm-config"])
-    assert llvm_config is not None
-    search_paths = run_llvm_config(["--bindir"]) + (
-        search_paths and (os.pathsep + search_paths)
-    )
+def _initialize(llvm_config: Optional[str] = None, search_paths: Optional[List[str]] = None):
```

Comment:
I'm a bit confused about how this is supposed to be used.
It seems like you're trying to be backward compatible, but we don't really care that much about it.

If you want to have make `Context` (optionally taking a path `llvm-config`) the entry point of the whole library, that you can instantiate against different `llvm-config`s, go ahead! Bump the version number to `0.2` and update `README.md`.

Reply:
> I'm a bit confused about how this is supposed to be used.

You either do (as you did previously)
```python
from llvmcpy.llvm import *
# "automatic mode", behaves the same as 0.1.6
get_global_context()
```
or
```python
from llvmcpy import get_wrapper
llvmcpy = get_wrapper("/path/to/llvm-config")
llvmcpy.get_global_context()
...
```

> If you want to have make Context ... the entry point of the whole library

I'm not sure, in the end the things you want to (optionally) provide are:
* The path to `llvm-config` (or an alternative name, e.g. `llvm-config-16`)
* A list of search path where to look for `$LLVM_CONFIG` or auxiliary executables such as `gcc`
and return a python module

I'd stick with `get_wrapper` because if feels the most KISS way to do it, however I'm open to suggestions

Reply:
> ```python
> # "automatic mode", behaves the same as 0.1.6
> ```

We don't care about backward compatibility.

> get_wrapper

It's not a "wrapper", it's just a context.
I'd rename `Context` to `LLVMCPy` and expose it to the user.

On a side note: I don't think you can load in a single process more than one LLVM library.

> ```python
> from llvmcpy import get_wrapper
> llvmcpy = get_wrapper("/path/to/llvm-config")
> llvmcpy.get_global_context()
> ...
> ```

This will first try to load the default `llvm-config` and then `/path/to/llvm-config`, correct?
This won't work, since you can't have two LLVM libraries in the same address space.

I think we should simplify things, have a single entry point with an optional path to `llvm-config` and no import-time stuff (also, no `__getattr__` stuff).

```python
from llvmcpy import LLVMCPy
llvmcpy = LLVMCPy("/path/to/llvm-config")
llvmcpy.get_global_context()
```

Simple and obvious.